components/toolbar/style.scss

```diff
@@ -47,11 +59,16 @@
 		font-size: 10px;
 		font-weight: bold;
 		position: relative;
-		top: -1px;
-		margin-left: -3px;
+		top: -5px;
+		left: -9px;
 	}
 }
 
+// compensate for making the button hit area include the visual spacing
+.components-toolbar__control.components-button + .components-toolbar__control.components-button {
+	margin-left: -3px;
```

Comment:
When the rule is not here, there's as much space for the button left as for the button right. With this rule, the entire spacing goes to the right button.